django/applications/catmaid/static/js/widgets/tagtable.js

```diff
@@ -0,0 +1,369 @@
+      "bAutoWidth": false,
+      "iDisplayLength": 30,
+      "aLengthMenu": [CATMAID.pageLengthOptions, CATMAID.pageLengthLabels],
+      "bJQueryUI": true,
+      "processing": true,
+      "deferRender": true,
+      "columns": [
+        {
+          "data": 'id',
+          "orderable": true,
+          "searchable": true,
+          "className": "center"
+        },
+        {
+          "data": 'tag',
+          "orderable": true,
+          "searchable": true,
+          "className": "center"
+        },
+        {
+          "data": 'skeletons',
+          "orderable": true,
+          "className": "center"
+        },
+        {
+          "data": 'checked',
+          "render": function(data, type, full) { return createCheckbox(data, type, full); },
+          "orderable": true,
+          "className": "center",
+          "width": "5%"
+        },
+        {
+          "data": 'nodes',
+          "orderable": true,
+          "className": "center"
+        }
+      ]
+    });
+
+    $(tableSelector + '_processing').show();
+
+    $(this.oTable).on('change', '.skelSelector', (function(event) {
+      var table = this.oTable.DataTable();
+      var row = table.row(event.currentTarget.closest('tr'));
+      var currentCheckedState = event.currentTarget.checked;
+      row.data().checked = currentCheckedState;
+      labelSkelMappingCache[row.data().id].selected = currentCheckedState;
+      row.invalidate();
+      if (currentCheckedState) {  // if checking box, just add
+        processSkelIDsFromLabelIDs([row.data().id], (function() {
+          if (row.data().checked) {
+            this.addAndSubtractFromSkeletonSource({
+              add: labelSkelMappingCache[row.data().id].skelIDs,
+              subtract: []
+            });
+          }
+        }).bind(this));
+      } else {  // if unchecking box, run full sync
+        this.syncSkeletonSource();
+      }
+    }).bind(this));
+
+    $(tableSelector + 'selectAllSkels').change((function(event){
+      // change all searched-for checkboxes to the same value as the header checkbox
+      var table = this.oTable.DataTable();
+      $(tableSelector + '_processing').show();  // doesn't show up immediately
+      table.rows({search: 'applied'}).every(function () {
+        // rows().every() may be slow, but is the only way to use search: 'applied'
+        // using rows().data() hits call stack limit with large data
+        var row = this;
+        var currentCheckedState = event.currentTarget.checked;
+        labelSkelMappingCache[row.data().id].selected = currentCheckedState;
+        if (row.data().checked != currentCheckedState) {
+          row.data().checked = currentCheckedState;
+          row.invalidate();
+        }
+      });
+
+      table.draw();
+      this.syncSkeletonSource();
+    }).bind(this));
+
+    $(tableSelector + "searchInputLabel").keydown((function (event) {
+      // filter table by tag text on hit enter
+      if (event.which == 13) {
```

Comment:
This was copied over from treenode-table, I think - what's the correct way to do it?